### PR TITLE
provide Mirage_runtime.register_arg, deprecate Mirage_runtime.register

### PR DESCRIPTION
**Pull request by hannesm**
The name `Mirage_runtime.register` is slightly too generic in my opinion.

### CODE REVIEW

File: lib_runtime/functoria/functoria_runtime.mli [hunk 1]
```diff
@@ -36,11 +36,18 @@ module Arg : sig
...
 (** [register t] registers the Cmdliner term [k] as a runtime key and return a
     callback [f] that evaluates to [t]s' value passed on the command-line.
 
     [f] will raise [Invalid_argument] if called before cmdliner's evaluation. *)
 
+val register_arg : 'a Cmdliner.Term.t -> unit -> 'a
+(** [register_arg t] registers the Cmdliner term [k] as a runtime key and return
+    a callback [f] that evaluates to [t]s' value passed on the command-line.
+
+    [f] will raise [Invalid_argument] if called before cmdliner's evaluation. *)
```
Comment (reynir) [inline lib_runtime/functoria/functoria_runtime.mli:49]: In the doc string we refer to it as (runtime) key, but we call the function `register_arg`. Should we maybe rename it to `register_key` or update the documentation?

**Comment by reynir** [comment]
While reviewing I notice that `Functoria_runtime.Arg` seems to be entirely unused outside of its parent module (and seems useless, too). Maybe we should just remove it from the .mli?

File: lib_runtime/functoria/functoria_runtime.mli (hunk 1 above)
Comment (hannesm) [reply lib_runtime/functoria/functoria_runtime.mli:49]: I guess we should decide on terminology ;)

I see in the changes entry for 4.5.0 the "configuration key" and "arguments at runtime".

Now, the module is called Key and Runtime_arg. I guess we should stick to that, then.

I find it very subtle, is it `key`, `parameter`, `argument`? Please also note that the docstring was only copied from the `register` function. But I'll adapt to "argument" to stay consistent.

**Comment by hannesm** [comment]
> While reviewing I notice that `Functoria_runtime.Arg` seems to be entirely unused outside of its parent module (and seems useless, too). Maybe we should just remove it from the .mli?

sounds like a good idea -- maybe best for another PR?

**Review by reynir — APPROVED** [review]
Looks good to me! Thanks

**Pull request merged**